### ISSUE TITLE
[BUG] Zoom level difference with MVT Layer

### ISSUE THREAD

**Issue by AnthonyGlt** [openlayers/openlayers#15653]
**Describe the bug**
When I add a MVT layer and a WMTS Layer of similar data, I notice that the details appear earlier in the WMTS than in the MVT.
By looking at the network panel, I've observed that we request 1 less zoom level in the MVT.

**To Reproduce**
I've added this MVT:
`https://data.geopf.fr/tms/1.0.0/PLAN.IGN/{z}/{x}/{y}.pbf` in 3857
And this WMTS:
```
url: 'https://data.geopf.fr/wmts',
layer:'GEOGRAPHICALGRIDSYSTEMS.PLANIGNV2',
matrixSet: 'PM',
format: 'image/png',
projection: ol.proj.get('EPSG:3857'),
```
This difference of zoom level is also present between MVT and XYZ

**Expected behavior**
I'm expecting to see the same level of data on both layer when I zoom in

**Comment by ahocevar** [comment]
This is expected, because the zoom levels are derived from the world width fitting a single tile. For MVT, the default is 512 pixels, for WMTS/XYZ 256. So what's zoom 0 on a MVT standard source is zoom 1 on a WMTS/XYZ standard source.

**Comment by AnthonyGlt** [comment]
Thank you for your answer. I'm trying to find some doc specifying the size of mvt tile. Is that a norm, is there a way to know which size is the MVT source ?
Thank you

**Comment by ahocevar** [comment]
Yes there is - for the source you linked, the metadata is here: https://data.geopf.fr/tms/1.0.0/PLAN.IGN/. And as you an see, the advertised tile size is 256x256px. So you need to configure the VectorTile source with `{tileSize: 256}`, because the default is `512`.

**Comment by AnthonyGlt** [comment]
Thank you, it seems that it was the olms lib (not updated) which was changing the size of the tile from 256 to 512. I'll try with a newer version.
Thanks !

**Comment by ahocevar** [comment]
You have to set the tileSize to 256 manually, because 512 is the default and ol-mapbox-style does not read TMS metadata.
